src/Summoner/Project.hs

```diff
@@ -0,0 +1,196 @@
+
+-- | Used for detecting the user decision during CLI input.
+data Decision = Yes | Nop | Idk
+
+instance Semigroup Decision where
+    (<>) :: Decision -> Decision -> Decision
+    Idk <> x   = x
+    x   <> Idk = x
+    _   <> x   = x
+
+instance Monoid Decision where
+    mempty  = Idk
+    mappend = (<>)
+
+data Targets = Targets
+    { githubFlag   :: Decision
+    , travisFlag   :: Decision
+    , appVeyorFlag :: Decision
+    , privateFlag  :: Decision
+    , scriptFlag   :: Decision
+    , isLibrary    :: Decision
+    , isExecutable :: Decision
+    , isTest       :: Decision
+    , isBenchmark  :: Decision
+    }
+
+instance Semigroup Targets where
+    t1 <> t2 = Targets
+        { githubFlag   = combine githubFlag
+        , travisFlag   = combine travisFlag
+        , appVeyorFlag = combine appVeyorFlag
+        , privateFlag  = combine privateFlag
+        , scriptFlag   = combine scriptFlag
+        , isLibrary    = combine isLibrary
+        , isExecutable = combine isExecutable
+        , isTest       = combine isTest
+        , isBenchmark  = combine isBenchmark
+        }
+      where
+        combine field = field t1 <> field t2
+
+instance Monoid Targets where
+    mempty  = Targets mempty mempty mempty mempty mempty mempty mempty mempty mempty
+    mappend = (<>)
+
+
+decisionToBool :: Decision -> Text -> IO Bool
+decisionToBool decision target = case decision of
+    Yes -> trueMessage  target
+    Nop -> falseMessage target
+    Idk -> do
+        ch <- choose ("Add " <> target <> "?") ["y", "n"]
+        case ch of
+            "y" -> trueMessage  target
+            "n" -> falseMessage target
+            _   -> error "Impossible happened"
+
+trueMessage, falseMessage :: Text -> IO Bool
+trueMessage  target = True  <$ successMessage (T.toTitle target <> " will be added to the project")
+falseMessage target = False <$ warningMessage (T.toTitle target <> " won't be added to the project")
+
+
+
+-- | Generate the project.
+generateProject :: Text -> Targets -> IO ()
+generateProject projectName Targets{..} = do
+    repo        <- checkUniqueName projectName
+    owner       <- queryDef "Repository owner: " defaultOwner
+    description <- query "Short project description: "
+    nm          <- queryDef "Author: " defaultName
+    email       <- queryDef "Maintainer e-mail: " defaultEmail
+    T.putStr
+        [text|
+        List of categories to choose from:
+
+          * Control                    * Concurrency
+          * Codec                      * Graphics
+          * Data                       * Sound
+          * Math                       * System
+          * Parsing                    * Network
+          * Text
+
+          * Application                * Development
+          * Compilers/Interpreters     * Testing
+          * Web
+          * Game
+          * Utility
+
+        |]
+    category <- query "Category: "
+    license  <- choose "License: " $ nub (defaultLicense : licenseNames)
```

Comment:
What do you think about adding `universum` to dependencies for `summoner`?

Reply:
Hmm, that will increase dependencies a bit..

Reply:
Yeah, that's right... When you have 123 vs. 124 dependency than having extra dependency is not really a problem if it increases productivity a lot! But since `summoner` is quite small let's wait for it to become bigger and see whether we can actually gain something from adding `universum`...